### PR TITLE
Fix worflow examples for java

### LINKED ISSUE #3864: Workflow examples are wrong for Java
**Describe the issue**

The `Task chaining`, `Fan-out/fan-in`,  `External system interaction` workflow Java examples are not correct. 

**URL of the docs**

Task chaining: https://docs.dapr.io/developing-applications/building-blocks/workflow/workflow-patterns/#task-chaining
Fan-out/fan-in: https://docs.dapr.io/developing-applications/building-blocks/workflow/workflow-patterns/#fan-outfan-in
External system interaction: https://docs.dapr.io/developing-applications/building-blocks/workflow/workflow-patterns/#external-system-interaction
**Expected content**

The Java example code should match the same logics in Python and Dotnet, but they are not. 

**Screenshots**

Take the `Task chaining` as an example, the below example code didn't show the correct logic. 
image

**Additional context**

### PR DESCRIPTION
**Pull request by kaibocai**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [X] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [X] Commands include options for Linux, MacOS, and Windows within codetabs
- [X] New file and folder names are globally unique
- [X] Page references use shortcodes instead of markdown or URL links
- [X] Images use HTML style and have alternative text
- [X] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This PR fixes the incorrect workflow Java example. 

## Issue reference

Resolve https://github.com/dapr/docs/issues/3864

### CODE REVIEW

**Comment by kaibocai** [comment]
Tag @cgillum , @johnewart , @cicoyle , @hhunter-ms for review.

File: daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md [hunk 1]
```diff
@@ -109,21 +109,53 @@ catch (TaskFailedException) // Task failures are surfaced as TaskFailedException
...
-  DaprWorkflowClient client = new DaprWorkflowClient();
+public class ChainWorkflow extends Workflow {
+    @Override
+    public WorkflowStub create() {
+        return ctx -> {
+            StringBuilder sb = new StringBuilder();
+            String wfInput = ctx.getInput(String.class);
+            String result1 = ctx.callActivity("event1", wfInput, String.class).await();
+            String result2 = ctx.callActivity("event2", result1, String.class).await();
+            String result3 = ctx.callActivity("event3", result2, String.class).await();
```
Comment (cgillum) [inline daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md]: Let's call these `Step`, `Step2`, and `Step3`, like we do with the other examples.

**Review by cgillum — COMMENTED** [review]
Thanks for making these great improvements! Just one nit-pick about naming to help make things a bit clearer and more consistent.

**Review by cgillum — APPROVED** [review]

**Review by hhunter-ms — APPROVED** [review]
thanks for your contribution!

**Comment by kaibocai** [comment]
@hhunter-ms , can you please help me merge this pr as I don't have permission to do it myself.  Thank you.

**Comment by hhunter-ms** [comment]
> can you please help me merge this pr as I don't have permission to do it myself. Thank you.

@kaibocai needed to fix some links first, will merge once it builds successfully!

**Pull request merged**